src/image_builder.py

```diff
@@ -154,109 +156,198 @@ def image_builder(buildspec):
+    return conclusion_stage_image_object
+
+def show_build_logs(images):
+
+    FORMATTER.title("Build Logs")
+
+    if not os.path.isdir("logs"):
+        os.makedirs("logs")
+
+    for image in images:
+        image_description = f"{image.name}-{image.stage}"
+        FORMATTER.title(image_description)
+        FORMATTER.table(image.info.items())
+        FORMATTER.separator()
+        FORMATTER.print_lines(image.log)
+        with open(f"logs/{image_description}", "w") as fp:
+            fp.write("/n".join(image.log))
+            image.summary["log"] = f"logs/{image_description}"
 
-        FORMATTER.title("Build Logs")
+def show_build_summary(images):
 
-        if not os.path.isdir("logs"):
-            os.makedirs("logs")
+    FORMATTER.title("Summary")
 
-        for image in IMAGES:
-            FORMATTER.title(image.name)
-            FORMATTER.table(image.info.items())
-            FORMATTER.separator()
-            FORMATTER.print_lines(image.log)
-            with open(f"logs/{image.name}", "w") as fp:
-                fp.write("/n".join(image.log))
-                image.summary["log"] = f"logs/{image.name}"
+    for image in images:
+        FORMATTER.title(image.name)
+        FORMATTER.table(image.summary.items())
 
-        FORMATTER.title("Summary")
+def show_build_errors(images):
+    FORMATTER.title("Errors")
+    is_any_build_failed = False
+    is_any_build_failed_size_limit = False
 
-        for image in IMAGES:
+    for image in images:
+        if image.build_status == constants.FAIL:
             FORMATTER.title(image.name)
-            FORMATTER.table(image.summary.items())
-
-        FORMATTER.title("Errors")
-        is_any_build_failed = False
-        is_any_build_failed_size_limit = False
-        for image in IMAGES:
-            if image.build_status == constants.FAIL:
-                FORMATTER.title(image.name)
-                FORMATTER.print_lines(image.log[-10:])
-                is_any_build_failed = True
-            else:
-                if image.build_status == constants.FAIL_IMAGE_SIZE_LIMIT:
-                    is_any_build_failed_size_limit = True
-        if is_any_build_failed:
-            raise Exception("Build failed")
+            FORMATTER.print_lines(image.log[-10:])
+            is_any_build_failed = True
+        else:
+            if image.build_status == constants.FAIL_IMAGE_SIZE_LIMIT:
+                is_any_build_failed_size_limit = True
+    if is_any_build_failed:
+        raise Exception("Build failed")
+    else:
+        if is_any_build_failed_size_limit:
+            FORMATTER.print("Build failed. Image size limit breached.")
         else:
-            if is_any_build_failed_size_limit:
-                FORMATTER.print("Build failed. Image size limit breached.")
+            FORMATTER.print("No errors")
+    return is_any_build_failed, is_any_build_failed_size_limit
+
+def upload_metrics(images, BUILDSPEC, is_any_build_failed, is_any_build_failed_size_limit):
```

Comment:
Please add a docstring to this function.

Reply:
Done.